apidocs: Migrate admin_config out of templates.

MSurfer20 opened the pull request:
Currently, the `admin_config` configuration was
hardcoded in the templates, but as a goal of creating
a common template, we need to move all configurations
outside.

Moved the checking for function and language which need
admin_config out of templates into the code.

<!-- What's this PR for?  (Just a link to an issue is fine.) -->


**Testing plan:** <!-- How have you tested? -->


**GIFs or screenshots:** <!-- If a UI change.  See:
  https://zulip.readthedocs.io/en/latest/tutorials/screenshot-and-gif-software.html
  -->


<!-- Also be sure to make clear, coherent commits:
  https://zulip.readthedocs.io/en/latest/contributing/version-control.html
  -->

MSurfer20 commented:
As always, checked the diffs as well. No differences :)

timabbott commented on zerver/openapi/markdown_extension.py:
```diff
@@ -76,6 +76,13 @@
+    "/users:post",
+    "/messages/{message_id}:delete",
+    "/realm/profile_fields:patch",
+]
```
I think structurally, whether an endpoint requires an organization administrator to use it will be a property of the endpoint, and thus should probably live in the OpenAPI files.  This is an OK intermediate state, but maybe we should add an `x-requires-administrator` tag using https://swagger.io/docs/specification/openapi-extensions/?

We will likely also want to have that flag control whether to show a "requires organization administrator" banner at the top of the page like we do in many Help Center articles.

MSurfer20 commented:
@timabbott Addressed the above comments. Attaching the diffs of the raw HTML here:
```
old-create-user new-create-user
215a216,218
> <p>This endpoint is only available to organization administrators.</p>
> </div>
> <div class="admonition warn">

old-deactivate-user new-deactivate-user
230c230
< <div class="codehilite" data-code-language="Python"><pre><span></span><code><span class="ch">#!/usr/bin/env python3</span>
---
> <div class="codehilite" data-code-language="Python"><pre><span></span><code><span class="ch">#!/usr/bin/env python</span>
234,235c234,235
< <span class="c1"># Pass the path to your zuliprc file here.</span>
< <span class="n">client</span> <span class="o">=</span> <span class="n">zulip</span><span class="o">.</span><span class="n">Client</span><span class="p">(</span><span class="n">config_file</span><span class="o">=</span><span class="s2">"~/zuliprc"</span><span class="p">)</span>
---
> <span class="c1"># The user for this zuliprc file must be an organization administrator</span>
> <span class="n">client</span> <span class="o">=</span> <span class="n">zulip</span><span class="o">.</span><span class="n">Client</span><span class="p">(</span><span class="n">config_file</span><span class="o">=</span><span class="s2">"~/zuliprc-admin"</span><span class="p">)</span>

old-delete-message new-delete-message
214a215,217
> <div class="admonition warn">
> <p>This endpoint is only available to organization administrators.</p>
> </div>

old-reactivate-user new-reactivate-user
230c230
< <div class="codehilite" data-code-language="Python"><pre><span></span><code><span class="ch">#!/usr/bin/env python3</span>
---
> <div class="codehilite" data-code-language="Python"><pre><span></span><code><span class="ch">#!/usr/bin/env python</span>
234,235c234,235
< <span class="c1"># Pass the path to your zuliprc file here.</span>
< <span class="n">client</span> <span class="o">=</span> <span class="n">zulip</span><span class="o">.</span><span class="n">Client</span><span class="p">(</span><span class="n">config_file</span><span class="o">=</span><span class="s2">"~/zuliprc"</span><span class="p">)</span>
---
> <span class="c1"># The user for this zuliprc file must be an organization administrator</span>
> <span class="n">client</span> <span class="o">=</span> <span class="n">zulip</span><span class="o">.</span><span class="n">Client</span><span class="p">(</span><span class="n">config_file</span><span class="o">=</span><span class="s2">"~/zuliprc-admin"</span><span class="p">)</span>

old-update-user new-update-user
215,217d214
< <div class="admonition warn">
< <p>This endpoint is only available to organization administrators.</p>
< </div>
```

timabbott commented on zerver/openapi/zulip.yaml:
```diff
@@ -5007,6 +5007,7 @@ paths:
         This API corresponds to the
         [delete a message completely][delete-completely] feature documented in
         the Zulip Help Center.
+      x-admin-config: true
```
Let's call this `x-requires-administrator: true`; I don't think `admin-config` is going to read particularly intuitively.

timabbott commented on zerver/openapi/markdown_extension.py:
```diff
@@ -525,6 +528,8 @@ def render_title(self, function: str) -> List[str]:
         title.extend(raw_title.splitlines())
         title = ["# " + line for line in title]
+        if check_admin_config(path, method):
+            title.append("{!api-admin-only.md!}")
```
This ends up on its own line because it's a list, right?

MSurfer20 replied:
Indeed -- the list contains each line that is supposed to be present( as can be seen in `title.extend(raw_title.splitlines())`)

timabbott commented on zerver/openapi/markdown_extension.py:
```diff
@@ -19,6 +19,7 @@
 
 import zerver.openapi.python_examples
 from zerver.openapi.openapi import (
+    check_admin_config,
```
`check_requires_administrator` would be a better name for this.

timabbott commented:
This looks great code-structure wise; posted a few comments on naming for better readability.

MSurfer20 commented:
@timabbott Thanks for the review. Fixed the naming issues :)

timabbott commented:
Merged, thanks @MSurfer20!
